consensus/event_loop.go

```diff
@@ -415,14 +417,135 @@ func (fc *FastHotStuffEventLoop) resetScheduledTasks() {
 	}
 
 	// Schedule the next block construction task. This will run with currentView param.
-	fc.nextBlockConstructionTask.Schedule(fc.blockConstructionInterval, fc.currentView, fc.onBlockConstructionScheduledTask)
+	fc.nextBlockConstructionTask.Schedule(fc.blockConstructionInterval, fc.currentView, fc.onBlockConstructionScheduledTaskExecuted)
 
 	// Schedule the next timeout task. This will run with currentView param.
 	fc.nextTimeoutTask.Schedule(timeoutDuration, fc.currentView, fc.onTimeoutScheduledTaskExecuted)
 }
 
-func (fc *FastHotStuffEventLoop) onBlockConstructionScheduledTask(blockConstructionView uint64) {
-	// TODO
+// When this function is triggered, it means that we have reached the block construction
+// time ETA for blockConstructionView. If we have a QC or timeout QC for the view, then we
+// signal the server.
+func (fc *FastHotStuffEventLoop) onBlockConstructionScheduledTaskExecuted(blockConstructionView uint64) {
```

Comment:
Hmm what if we can't construct a block after the blockConstructionInterval, but then we recieve a couple more votes and now we're able to? Would `onBlockConstructionScheduledTaskExecuted` be called again? Also, should this be gated by a check that we're the leader for this view?

Reply:
The `onBlockConstructionScheduledTaskExecuted` will get rescheduled on its interval and reruns. See the next PR for the reschedule. 

Re: the leader check, the event loop doesn't really know when the node is leader. This keeps the event loop as simple as possible. The server will know when the node is the leader though. This separation is fine because:
1. The server will only pass vote and timeout messages to the event loop for views when it will be the leader
2. The server will only accept and act on the block construction signal for views when it is the leader

Reply:
Ah yes, I saw you changed it in the next PR. Moved my comments there. And makes sense on the leader check!